src/PaymentGateways/Gateways/Stripe/Workflow.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Give\PaymentGateways\Gateways\Stripe;
+
+use ReflectionMethod;
+
+/**
+ * Encapsulate sequential processes with a shared context.
+ */
+class Workflow
+{
+    protected $container = [];
+
+    public function __construct()
+    {
+        $this->bind( $this );
+    }
+
+    public function bind( $concrete ) {
+        $this->container[ get_class( $concrete ) ] = $concrete;
+        return $this;
+    }
+
+    public function resolve( $abstract ) {
+        if( ! isset( $this->container[ $abstract ] ) ) {
+            throw new \Exception( "Abstract $abstract not found." );
+        }
+        return $this->container[ $abstract ];
+    }
+
+    public function action( callable $action )
+    {
+        if( is_a( $action, WorkflowAction::class ) ) {
+            $action->attachWorkflow( $this );
+        }
+
+        $reflection = new ReflectionMethod($action, '__invoke');
+        $action( ...array_map( function( $parameter ) {
+            return $this->resolve( $parameter->getType()->getName() );
+        }, $reflection->getParameters() ) );
+
+        return $this;
+    }
```

Comment:
@kjohnson sorry bud

<img width="829" alt="Screen Shot 2022-01-28 at 10 45 47 AM" src="https://user-images.githubusercontent.com/10138447/151581259-666aae51-86ab-480a-912d-50ca4de58421.png">

Reply:
Hm... I probably just need to access it differently. We are using a container in Give, so I'll look there and update.

Reply:
Found it. I'll update my implementation. Thanks for catching this!

```php
        // Use ReflectionParameter::getClass() prior to its replacement in PHP 7.1
        if (version_compare(PHP_VERSION, '7.1', '<')) {
            $class = $parameter->getClass();

            return $class ? $class->name : null;
        }

        return $parameter->hasType() ? $parameter->getType()->getName() : null;
```

Reply:
@jonwaldstein Resolved by 9976f98.